Managers/ConsoleMan.cpp

```diff
@@ -1,541 +1,423 @@
+			if (IsEnabled()) {
+				if (!m_ReadOnly) {
+					m_InputTextBox->SetEnabled(false);
+					m_GUIController->GetManager()->SetFocus(0);
+					// Save any text being worked on in the input, as the box keeps getting junk added to it
+					m_LastInputString = m_InputTextBox->GetText();
+					SetEnabled(false);
+				} else {
+					// Restore any text being worked on in the input and set the focus and cursor position of the input line
+					m_InputTextBox->SetText(m_LastInputString);
+					m_InputTextBox->SetCursorPos(m_InputTextBox->GetText().length());
+					m_ReadOnly = false;
+				}
+			} else {
+				m_InputTextBox->SetText(m_LastInputString);
+				m_InputTextBox->SetCursorPos(m_InputTextBox->GetText().length());
+				SetEnabled(true);
+			}
+		}
+
+		if (m_ConsoleState != ENABLED && m_ConsoleState != DISABLED) { ConsoleOpenClose(); }
+
+		if (m_ConsoleState != ENABLED) {
+			return;
+		}
+
+		m_GUIController->Update();
+
+		// Resize console using Ctrl+Up/Down
+		if (g_UInputMan.FlagCtrlState() && g_UInputMan.KeyPressed(KEY_DOWN)) {
+			SetConsoleScreenSize(m_ConsoleScreenRatio + 0.05F);
+		} else if (g_UInputMan.FlagCtrlState() && g_UInputMan.KeyPressed(KEY_UP)) {
+			SetConsoleScreenSize(m_ConsoleScreenRatio - 0.05F);
+		}
+
+		if (!m_ReadOnly) {
+			m_InputTextBox->SetEnabled(true);
+			m_InputTextBox->SetFocus();
+	
+			// Get rid of any grave accents (`) that are pasted/typed into the textbox.
+			std::string textBoxString = m_InputTextBox->GetText();
+			if (std::find(textBoxString.begin(), textBoxString.end(), '`') != textBoxString.end()) {
+				textBoxString.erase(std::remove(textBoxString.begin(), textBoxString.end(), '`'), textBoxString.end());
+				m_InputTextBox->SetText(textBoxString);
+				m_InputTextBox->SetCursorPos(textBoxString.length());
+			}	
+		} else {
+			m_InputTextBox->SetEnabled(false);
+			m_GUIController->GetManager()->SetFocus(0);
+			return;
+		}	
+
+		if (g_UInputMan.KeyPressed(KEY_ENTER) || g_UInputMan.KeyPressed(KEY_ENTER_PAD)) { FeedString(); }
+
+		// If multiple strings were pasted into the box, we need to check for and submit all the strings
+		if (m_InputTextBox->GetText().find_last_of('\n') != std::string::npos) { FeedMultipleStrings(); }
+
+		if (!m_InputLog.empty() && !g_UInputMan.FlagCtrlState()) {
+			if (g_UInputMan.KeyPressed(KEY_UP)) {
+				LoadLoggedInput(false);
+			} else if (g_UInputMan.KeyPressed(KEY_DOWN)) {
+				LoadLoggedInput(true);
+			}
+		}
+
+		// TODO: Get this working and see if it actually makes any difference.
+		/*
+		// Cut off the text in the text label at a reasonable height so it doesn't get really slow to draw
+		if (m_ConsoleText->GetTextHeight() > g_FrameMan.GetResY() {
+		}
+		*/
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ConsoleMan::FeedString() {
+		// Clear the errors from the VM so we will get fresh feedback
```

Comment:
All the comments in here except for the one about only saving input if it's different are useless.
I think you could probably pull that useful one into the method comment instead, i.e.
```
/// <summary>
/// Executes the input string when Enter is pressed, saves it to the input log if it's different from the previous string, and clears the textbox. This is called from Update().
/// </summary>
```